scipy/optimize/minpack.py

```diff
@@ -510,28 +523,41 @@ def curve_fit(f, xdata, ydata, p0=None, sigma=None, absolute_sigma=False,
         values will all be 1 (if the number of parameters for the function
         can be determined using introspection, otherwise a ValueError
         is raised).
-    sigma : None or M-length sequence, optional
-        If not None, the uncertainties in the ydata array. These are used as
-        weights in the least-squares problem
-        i.e. minimising ``np.sum( ((f(xdata, *popt) - ydata) / sigma)**2 )``
-        If None, the uncertainties are assumed to be 1.
+    sigma : None or M-length sequence or MxM array, optional
+        Determines the errors in `ydata`. If we define residuals as
```

Comment:
My understanding "uncertainty = covariance matrix of errors". Thus at the beginning we write "uncertainty", but then we get more specific: "standard deviations of errors in `ydata`" and "the covariance matrix of errors in `ydata`". 

So my original comment was related to changing "of uncertainties in `ydata`" to "of errors in `ydata`" in the list.